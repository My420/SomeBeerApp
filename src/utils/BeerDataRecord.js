import { Record } from 'immutable';

const BeerDataRecord = Record(
  {
    id: null,
    name: null,
    tagline: null,
    first_brewed: null,
    description: null,
    image_url: null,
    abv: null,
    ibu: null,
    target_fg: null,
    target_og: null,
    ebc: null,
    srm: null,
    ph: null,
    attenuation_level: null,
    volume: {},
    boil_volume: {},
    method: {},
    ingredients: {},
    food_pairing: [],
    brewers_tips: null,
    contributed_by: null
  },
  'BeerDataRecord'
);

export default BeerDataRecord;
