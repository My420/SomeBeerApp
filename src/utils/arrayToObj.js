/* eslint no-param-reassign: "error" */

const arrayToObj = arr => {
  return arr.reduce((prevValue, element) => {
    prevValue[element.id] = element;
    return prevValue;
  }, {});
};

export default arrayToObj;
