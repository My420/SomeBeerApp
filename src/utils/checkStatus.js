const checkStatus = response => {
  const responseStatus = +response.status;
  if (responseStatus >= 200 && responseStatus < 300) {
    return response;
  }
  throw new Error(response.statusText);
};

export default checkStatus;
