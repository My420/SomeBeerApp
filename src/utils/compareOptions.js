const compareOptions = (opt1, opt2) => {
  let isEqual = true;
  Object.keys(opt1).forEach(value => {
    if (opt1[value] !== opt2[value]) {
      isEqual = false;
    }
  });
  return isEqual;
};

export default compareOptions;
