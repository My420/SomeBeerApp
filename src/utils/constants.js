export const SHOP_NAME = 'Some Beer Shop';

export const LINK = {
  HOME: '/',
  CATALOG: '/catalog',
  ROULETTE: '/roulette',
  ABOUT: '/about',
  FAVORITE: '/favorite',
  CART: '/cart'
};

export const NAVIGATION_LINKS = [
  [LINK.HOME, 'Home'],
  [LINK.CATALOG, 'Catalog'],
  [LINK.ROULETTE, 'Roulette!'],
  [LINK.FAVORITE, 'Favorite'],
  [LINK.ABOUT, 'About'],
  [LINK.CART, 'Cart']
];

export const BROWSER_WIDTH_RATIO = {
  LPC: 4,
  MPC: 3,
  TABLET_9: 2,
  TABLET_7: 1,
  MOBILE: 0
};

export const BROWSER_MIN_WIDTH = {
  LPC: 1200,
  MPC: 993,
  TABLET_9: 768,
  TABLET_7: 480,
  MOBILE: 320
};

export const SOCIAL_LINK = {
  GITHUB: 'https://github.com/My420',
  EMAIL: 'mailto:klinovitsky.aleksey@gmail.com',
  SKYPE: 'skype:live:klinovitsky.aleksey?chat'
};

export const AUTHOR = 'Klinovitsky Aleksey';

export const POPULAR_BEER__GET_REQUEST = 'POPULAR_BEER__GET_REQUEST';
export const POPULAR_BEER__GET_SUCCESS = 'POPULAR_BEER__GET_SUCCESS';
export const POPULAR_BEER__GET_FAILURE = 'POPULAR_BEER__GET_FAILURE';

export const API_HOST = 'https://api.punkapi.com/v2/';

export const POPULAR_BEER_AMOUNT = 8;

export const LOAD_ERROR_MESSAGE =
  'Could not download data, please refresh the page';

export const ERROR_TITLE = 'Oops! Something went wrong.';
export const ERROR_ACTION = 'Please refresh page or try again later.';
