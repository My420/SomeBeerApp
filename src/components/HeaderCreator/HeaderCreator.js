import React from 'react';
import PropTypes from 'prop-types';
import Header from '../Header/Header';
import withResize from '../ResizeContext/ResizeWrapper';
import withToggle from '../HOC/ToggleWrapper/ToggleWrapper';
import { BROWSER_WIDTH_RATIO } from '../../utils/constants';

export const HeaderCreator = ({ browserWidthRatio }) => {
  // eslint-disable-next-line no-console
  console.log('render ===== HeaderCreator');
  const HeaderComponent = withToggle(
    Header,
    !(browserWidthRatio < BROWSER_WIDTH_RATIO.MPC)
  );
  return (
    <HeaderComponent
      isButtonShown={browserWidthRatio < BROWSER_WIDTH_RATIO.MPC}
    />
  );
};

HeaderCreator.propTypes = {
  browserWidthRatio: PropTypes.oneOf([
    BROWSER_WIDTH_RATIO.MOBILE,
    BROWSER_WIDTH_RATIO.TABLET_7,
    BROWSER_WIDTH_RATIO.TABLET_9,
    BROWSER_WIDTH_RATIO.MPC,
    BROWSER_WIDTH_RATIO.LPC
  ]).isRequired
};

export default withResize(HeaderCreator);
