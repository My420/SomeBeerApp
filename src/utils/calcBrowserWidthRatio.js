import { BROWSER_MIN_WIDTH, BROWSER_WIDTH_RATIO } from './constants';

const calcBrowserWidthRatio = width => {
  if (width > BROWSER_MIN_WIDTH.LPC) {
    return BROWSER_WIDTH_RATIO.LPC;
  }
  if (width > BROWSER_MIN_WIDTH.MPC) {
    return BROWSER_WIDTH_RATIO.MPC;
  }
  if (width > BROWSER_MIN_WIDTH.TABLET) {
    return BROWSER_WIDTH_RATIO.TABLET_9;
  }
  if (width > BROWSER_MIN_WIDTH.MOBILE) {
    return BROWSER_WIDTH_RATIO.TABLET_7;
  }
  return BROWSER_WIDTH_RATIO.MOBILE;
};

export default calcBrowserWidthRatio;
