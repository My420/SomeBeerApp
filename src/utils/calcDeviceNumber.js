import { DEVICE_MIN_WIDTH, DEVICE_NUMBER } from './constants';

const calcDeviceNumber = width => {
  if (width > DEVICE_MIN_WIDTH.LPC) {
    return DEVICE_NUMBER.LPC;
  }
  if (width > DEVICE_MIN_WIDTH.MPC) {
    return DEVICE_NUMBER.MPC;
  }
  if (width > DEVICE_MIN_WIDTH.TABLET) {
    return DEVICE_NUMBER.TABLET;
  }
  return DEVICE_NUMBER.MOBILE;
};

export default calcDeviceNumber;
