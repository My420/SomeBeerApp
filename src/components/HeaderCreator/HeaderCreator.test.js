import React from 'react';
import { shallow } from 'enzyme';
import { HeaderCreator } from './HeaderCreator';
import { BROWSER_WIDTH_RATIO } from '../../utils/constants';

describe('Test <HeaderCreator /> component', () => {
  describe('prop browserWidthRatio = BROWSER_WIDTH_RATIO.LPC', () => {
    const wrapper = shallow(
      <HeaderCreator browserWidthRatio={BROWSER_WIDTH_RATIO.LPC} />
    );

    test('should correct render <HeaderCreator /> component ', () => {
      expect(wrapper).toMatchSnapshot();
    });

    test('should render <ToggleWrapper(Header)/>', () => {
      expect(wrapper.find('ToggleWrapper(Header)')).toHaveLength(1);
    });

    test(' <ToggleWrapper(Header)/> should have props isButtonShown = false', () => {
      expect(
        wrapper.find('ToggleWrapper(Header)').prop('isButtonShown')
      ).toBeFalsy();
    });
  });

  describe('prop browserWidthRatio = BROWSER_WIDTH_RATIO.MPC', () => {
    const wrapper = shallow(
      <HeaderCreator browserWidthRatio={BROWSER_WIDTH_RATIO.MPC} />
    );

    test('should correct render <HeaderCreator /> component ', () => {
      expect(wrapper).toMatchSnapshot();
    });

    test('should render <ToggleWrapper(Header)/>', () => {
      expect(wrapper.find('ToggleWrapper(Header)')).toHaveLength(1);
    });

    test(' <ToggleWrapper(Header)/> should have props isButtonShown = false', () => {
      expect(
        wrapper.find('ToggleWrapper(Header)').prop('isButtonShown')
      ).toBeFalsy();
    });
  });

  describe('prop browserWidthRatio = BROWSER_WIDTH_RATIO.TABLET_9', () => {
    const wrapper = shallow(
      <HeaderCreator browserWidthRatio={BROWSER_WIDTH_RATIO.TABLET_9} />
    );

    test('should correct render <HeaderCreator /> component ', () => {
      expect(wrapper).toMatchSnapshot();
    });

    test('should render <ToggleWrapper(Header)/>', () => {
      expect(wrapper.find('ToggleWrapper(Header)')).toHaveLength(1);
    });

    test(' <ToggleWrapper(Header)/> should have props isButtonShown = true', () => {
      expect(
        wrapper.find('ToggleWrapper(Header)').prop('isButtonShown')
      ).toBeTruthy();
    });
  });

  describe('prop browserWidthRatio = BROWSER_WIDTH_RATIO.TABLET_7', () => {
    const wrapper = shallow(
      <HeaderCreator browserWidthRatio={BROWSER_WIDTH_RATIO.TABLET_7} />
    );

    test('should correct render <HeaderCreator /> component ', () => {
      expect(wrapper).toMatchSnapshot();
    });

    test('should render <ToggleWrapper(Header)/>', () => {
      expect(wrapper.find('ToggleWrapper(Header)')).toHaveLength(1);
    });

    test(' <ToggleWrapper(Header)/> should have props isButtonShown = true', () => {
      expect(
        wrapper.find('ToggleWrapper(Header)').prop('isButtonShown')
      ).toBeTruthy();
    });
  });

  describe('prop browserWidthRatio = BROWSER_WIDTH_RATIO.MOBILE', () => {
    const wrapper = shallow(
      <HeaderCreator browserWidthRatio={BROWSER_WIDTH_RATIO.MOBILE} />
    );

    test('should correct render <HeaderCreator /> component ', () => {
      expect(wrapper).toMatchSnapshot();
    });

    test('should render <ToggleWrapper(Header)/>', () => {
      expect(wrapper.find('ToggleWrapper(Header)')).toHaveLength(1);
    });

    test(' <ToggleWrapper(Header)/> should have props isButtonShown = true', () => {
      expect(
        wrapper.find('ToggleWrapper(Header)').prop('isButtonShown')
      ).toBeTruthy();
    });
  });
});
