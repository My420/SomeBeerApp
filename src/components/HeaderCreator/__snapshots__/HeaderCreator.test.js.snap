// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test <HeaderCreator /> component prop browserWidthRatio = BROWSER_WIDTH_RATIO.LPC should correct render <HeaderCreator /> component  1`] = `
<ToggleWrapper(Header)
  isButtonShown={false}
/>
`;

exports[`Test <HeaderCreator /> component prop browserWidthRatio = BROWSER_WIDTH_RATIO.MOBILE should correct render <HeaderCreator /> component  1`] = `
<ToggleWrapper(Header)
  isButtonShown={true}
/>
`;

exports[`Test <HeaderCreator /> component prop browserWidthRatio = BROWSER_WIDTH_RATIO.MPC should correct render <HeaderCreator /> component  1`] = `
<ToggleWrapper(Header)
  isButtonShown={false}
/>
`;

exports[`Test <HeaderCreator /> component prop browserWidthRatio = BROWSER_WIDTH_RATIO.TABLET_7 should correct render <HeaderCreator /> component  1`] = `
<ToggleWrapper(Header)
  isButtonShown={true}
/>
`;

exports[`Test <HeaderCreator /> component prop browserWidthRatio = BROWSER_WIDTH_RATIO.TABLET_9 should correct render <HeaderCreator /> component  1`] = `
<ToggleWrapper(Header)
  isButtonShown={true}
/>
`;
